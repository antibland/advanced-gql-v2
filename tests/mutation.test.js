const gql = require("graphql-tag");
const createTestServer = require("./helper");
const UPDATE_USER_INPUT = gql`
  mutation {
    updateMe(input: { email: "some_new_email@email.com" }) {
      email
    }
  }
`;

describe("mutations", () => {
  test("update user input", async () => {
    const { mutate } = await createTestServer({
      user: { id: 1 },
      models: {
        User: {
          updateOne() {
            return {
              email: "some_new_email@email.com",
            };
          },
        },
        user: { id: 1 },
      },
    });
    const res = await mutate({ query: UPDATE_USER_INPUT });
    expect(res).toMatchSnapshot();
  });
});
