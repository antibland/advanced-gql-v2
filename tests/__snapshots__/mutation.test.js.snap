// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mutations update user input 1`] = `
Object {
  "data": Object {
    "updateMe": Object {
      "email": "some_new_email@email.com",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
